src/EditorFeatures/CSharpTest/CodeActions/NamedArguments/AddNamedArgumentsTests.cs

```diff
@@ -0,0 +1,141 @@
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.NamedArguments;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.NamedArguments
+{
+    public class AddNamedArgumentsTests : AbstractCSharpCodeActionTest
+    {
+        protected override object CreateCodeRefactoringProvider(Workspace workspace)
+        {
+            return new CSharpAddNamedArgumentsCodeRefactoringProvider();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestConditionalMethod()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, int arg2) => this?.[|M|](1, 2); }",
+@"class C { void M(int arg1, int arg2) => this?.M(arg1: 1, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestConditionalIndexer()
+        {
+            await TestAsync(
+@"class C { int? this[int arg1, int arg2] => this?[|[1, 2]|]; }",
+@"class C { int? this[int arg1, int arg2] => this?[arg1: 1, arg2: 2]; }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestThisConstructorInitializer()
+        {
+            await TestAsync(
+@"class C { C(int arg1, int arg2) {} C() : [|this(1, 2)|] {} }",
+@"class C { C(int arg1, int arg2) {} C() : this(arg1: 1, arg2: 2) {} }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestBaseConstructorInitializer()
+        {
+            await TestAsync(
+@"class C { public C(int arg1, int arg2) {} } class D : C { D() : [|base(1, 2)|] {} }",
+@"class C { public C(int arg1, int arg2) {} } class D : C { D() : base(arg1: 1, arg2: 2) {} }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestConstructor()
+        {
+            await TestAsync(
+@"class C { C(int arg1, int arg2) { new [|C|](1, 2); } }",
+@"class C { C(int arg1, int arg2) { new C(arg1: 1, arg2: 2); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestIndexer()
+        {
+            await TestAsync(
+@"class C { char M(string arg1) => [|arg1|][0]; }",
+@"class C { char M(string arg1) => arg1[index: 0]; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestArrayIndexer()
+        {
+            await TestMissingAsync(
+@"class C { int M(int[] arg1) => [|arg1|][0]; }");
+
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingArgumentNames()
+        {
```

Comment:
Nit: From the name of the test, I thought the refactoring would be missing in this scenario, but that doesn't seem to be the case.. In the IDE code base, we generally prefix something like `TestMissing` when we imply the refactoring should be missing in this case.